Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -411,13 +418,21 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
 
         StringBuilder builder;
         auto first = true;
-        for (auto& value : m_properties.values) {
+        for (size_t i = 0; i < m_properties.values.size(); ++i) {
+            auto value = m_properties.values[i];
+            auto value_string = value->to_string(mode);
+            auto initial_value_string = property_initial_value(m_properties.sub_properties[i])->to_string(mode);
+            if (value_string == initial_value_string && !value->is_css_wide_keyword() && value->to_keyword() != Keyword::Auto)
```

Comment:
I'm a little confused by these exceptions:
- The initial value will never be a css-wide keyword, so that's excluded by the first comparison.
- Why do we need to always serialize `auto`?

Reply:
I have removed both of these exceptions. The CSS wide keyword check is completely unnecessary, as you say. The `auto` check certainly isn't a general rule that should apply to everything; not having it leads to a small regression in `inset` property serialization. This would be better handled by a special case. I'd rather do that as a separate PR, because it looks like a few changes to `inset` are needed to make it more spec-compliant.